modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/config/BaseConfiguration.java

```diff
@@ -107,8 +105,8 @@ public void build() throws AxisFault {
         httpParams = buildHttpParams();
         ioReactorConfig = buildIOReactorConfig();
         String sysCorrelationStatus = System.getProperty(PassThroughConstants.CORRELATION_LOGS_SYS_PROPERTY);
-        if (sysCorrelationStatus != null) {
-            correlationLoggingEnabled = sysCorrelationStatus.equalsIgnoreCase("true");
+        if (sysCorrelationStatus != null && !PassThroughCorrelationConfigDataHolder.isEnable()) {
```

Comment:
what is the purpose of this check '!PassThroughCorrelationConfigDataHolder.isEnable()'

Reply:
If the dataholder is updated through the log service, base configuration will not reset the dataholder value again